compiler/rustc_typeck/src/check/op.rs

```diff
@@ -683,14 +637,10 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                         format!("cannot apply unary operator `{}`", op.as_str()),
                     );
 
-                    let mut visitor = TypeParamVisitor(vec![]);
-                    visitor.visit_ty(operand_ty);
-                    if let [_] = &visitor.0[..] && let ty::Param(_) = *operand_ty.kind() {
```

Comment:
This is technically more general now, since this only matches a type parameter exactly. I found no changes in UI tests when changing it to `has_param_types_or_consts`.